src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -503,4 +504,116 @@ public void replaceVariablesInFile(String filePath, List<String> targetStrings,
         }
         Files.write(replaceFilePath, fileContent.getBytes(charset));
     }
+
+    /**
+     * This normalizes all line endings to UNIX-line-endings recursively from the startPath.
+     *
+     * @see {@link #normalizeLineEndings(String) normalizeLineEndings}
+     * @param startPath          the path where the start directory is located
+     * @throws IOException if an issue occurs on file access for the normalizing of the line endings.
+     */
+    public void normalizeLineEndingsRecursive(String startPath) throws IOException {
+        log.debug("Normalizing file endings in directory {}", startPath);
+        File directory = new File(startPath);
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new RuntimeException("File endings in directory " + startPath + " should be normalized but the directory does not exist.");
+        }
+
+        // Get all files in directory
+        String[] files = directory.list((current, name) -> new File(current, name).isFile());
```

Comment:
This method can return null under certain circumstances. I would propose using propose using FileUtils here: https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#listFiles(java.io.File,%20org.apache.commons.io.filefilter.IOFileFilter,%20org.apache.commons.io.filefilter.IOFileFilter)

Reply:
Then you would also get the recursive look-up for free.